airflow/models.py

```diff
@@ -1996,7 +1996,14 @@ def execute(self, context):
 
         Refer to get_template_context for more context.
         """
-        raise NotImplementedError()
+        if self.dag:
+            os.environ['AIRFLOW_DAG_ID'] = self.dag.dag_id
+        dagrun = context['dag_run']
```

Comment:
the context is prepared here:
https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L1424
and documented here:
http://pythonhosted.org/airflow/code.html#macros

You'll see that you can use `context['execution_date']` straight up